node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java

```diff
@@ -629,8 +629,10 @@ public Node markNodeAvailableForNewAllocation(String hostname, Agent agent, Stri
                 List<Node> removed = new ArrayList<>(children);
                 if (zone.cloud().value().equals("aws"))
                     db.removeNodes(List.of(node));
-                else
+                else {
+                    node = node.with(IP.Config.EMPTY);
```

Comment:
Consider extending one of the deprovision tests in `NodeRepositoryTest` to test this behaviour.